test/ssr/ssr-env.spec.js

```diff
@@ -0,0 +1,52 @@
+import Vue from '../../dist/vue.common.js'
+import createRenderer from '../../dist/server-renderer.js'
+import '../helpers/to-have-been-warned.js'
+
+describe('SSR: VUE_ENV=server', () => {
+  it('_isServer set as "server" on Vue config', () => {
+    const vm = new Vue({
+      data: {
+        foo: 'server',
+        bar: 'rendering'
+      }
+    })
+    expect(vm.$isServer).toBe(true)
+  })
+
+  it('$isServer set as true on VM', () => {
+    const vm = new Vue({
+      data: {
+        foo: 'server',
+        bar: 'rendering'
+      }
+    })
+    expect(vm.$isServer).toBe(true)
+  })
+
+  it('no data observations', () => {
+    const vm = new Vue({
+      data: {
+        foo: 'server',
+        bar: 'rendering'
+      },
+      computed: {
+        combined () {
+          return this.foo + this.bar
+        }
+      }
+    })
+
+    vm.foo = ''
+
+    expect(vm.foo).toBe('')
+    expect(vm.combined).toBe('rendering')
+    expect(vm.__ob__).toBe(undefined)
```

Comment:
Should be asserting `vm.$data.__ob__` here since Vue instances themselves are never observed.